hazelcast-jet-core/src/main/java/com/hazelcast/jet/Traversers.java

```diff
@@ -34,14 +34,11 @@ private Traversers() {
     }
 
     /**
-     * Returns an empty traverser equivalent to {@code () -> null}, but with
-     * guaranteed unique identity. Such a traverser is useful in a <em>null
-     * object</em> pattern, which relies on identity checks with {@code ==}
-     * against a private object.
+     * Returns a traverser that always returns null.
      */
     @Nonnull
-    public static <T> Traverser<T> newNullTraverser() {
-        return new NullTraverser<>();
+    public static <T> Traverser<T> nullTraverser() {
```

Comment:
This should be called `empty()`, to be called as `Traversers.empty()`. The name `newNullTraverser()` was used in reference to the "NULL object" pattern, which this doesn't serve.